### PR TITLE
[RHPAM-3331] Operator for AMQ Streams (Kafka) integration

### PR DESCRIPTION
**Pull request by desmax74**
See https://issues.redhat.com/browse/RHPAM-3331
Signed-off-by: desmax74 <mdessi@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [x] Pull Request title is properly formatted: `[RHPAM-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Attached commits represent units of work and are properly formatted
- [x] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [x] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

File: kieserver/image.yaml [hunk 1]
```diff
@@ -199,7 +199,31 @@ envs:
     description: "Disable Maven pull dependencies for immutable KIE Server configurations for S2I and pre built kjars. Useful for pre-compiled kjar."
   - name: "KIE_SERVER_DISABLE_KC_VERIFICATION"
     example: "true"
-    description: "Disable Maven KIE Jar verification. It is recommended to test the kjar manually before disabling this verification."
+    description: "Disable Maven KIE Jar verification. It is recommended to test the kjar manually before disabling this verification."  
+  - name: "org.kie.kafka.server.ext.disabled"
```
Comment (spolti) [inline kieserver/image.yaml:203]: shouldn't the env name be here and the system property on ks image?

**Review by spolti — CHANGES_REQUESTED** [review]

File: kieserver/image.yaml (hunk 1 above)
Comment (desmax74) [reply kieserver/image.yaml]: Updated

File: kieserver/image.yaml [hunk 2]
```diff
@@ -199,7 +199,31 @@ envs:
     description: "Disable Maven pull dependencies for immutable KIE Server configurations for S2I and pre built kjars. Useful for pre-compiled kjar."
   - name: "KIE_SERVER_DISABLE_KC_VERIFICATION"
     example: "true"
-    description: "Disable Maven KIE Jar verification. It is recommended to test the kjar manually before disabling this verification."
+    description: "Disable Maven KIE Jar verification. It is recommended to test the kjar manually before disabling this verification."  
+  - name: "KIE_KAFKA_EXT_DISABLED"
+    example: "true"
+    description: "Disable Kafka extension"
```
Comment (spolti) [inline kieserver/image.yaml:205]: Improve the description with more details and also which system property it sets.
e.g.:
https://github.com/jboss-container-images/rhpam-7-openshift-image/pull/524/files#diff-9c3271200df1057c7c0e0dd7542d17d0e2467adf63351fbb5cd6d51efe9b6062R140

Comment (spolti) [reply kieserver/image.yaml:205]: same for the others.

**Review by spolti — CHANGES_REQUESTED** [review]

File: kieserver/image.yaml (hunk 2 above)
Comment (spolti) [reply kieserver/image.yaml:205]: And, when it does have a default value, specify it as well.

Comment (desmax74) [reply kieserver/image.yaml:205]: ok

File: kieserver/image.yaml [hunk 3]
```diff
@@ -200,12 +200,12 @@ envs:
   - name: "KIE_SERVER_DISABLE_KC_VERIFICATION"
     example: "true"
     description: "Disable Maven KIE Jar verification. It is recommended to test the kjar manually before disabling this verification."  
-  - name: "KIE_KAFKA_EXT_DISABLED"
+  - name: "KIE_KAFKA_EXT_ENABLED"
     example: "true"
     description: "Disable Kafka extension"
   - name: "KIE_KAFKA_EXT_TOPICS"
     example: "event=events,book=books"
```
Comment (spolti) [inline kieserver/image.yaml]: is this description correct? seems to be misaligned with what we discussed today earlier.

Comment (desmax74) [reply kieserver/image.yaml]: Is correct, here the user just add the mapping and then there https://github.com/jboss-container-images/jboss-kie-modules/pull/446/files#diff-3e7628a0c0671f1c47f2f4da279754577748d791191318fb85c3d250b1ca182cR741  we add the syprop on every topic in the array

**Review by spolti — APPROVED** [review]

File: kieserver/image.yaml [hunk 4]
```diff
@@ -199,7 +199,31 @@ envs:
     description: "Disable Maven pull dependencies for immutable KIE Server configurations for S2I and pre built kjars. Useful for pre-compiled kjar."
   - name: "KIE_SERVER_DISABLE_KC_VERIFICATION"
     example: "true"
-    description: "Disable Maven KIE Jar verification. It is recommended to test the kjar manually before disabling this verification."
+    description: "Disable Maven KIE Jar verification. It is recommended to test the kjar manually before disabling this verification."  
+  - name: "KIE_SERVER_KAFKA_EXT_ENABLED"
+    example: "true"
+    description: "Setting with true enable the others KIE_KAFKA_EXT environments used by Kafka, this env will be used by system prop org.kie.kafka.server.ext.disable with the opposite value"
+  - name: "KIE_SERVER_KAFKA_EXT_TOPICS"
```
Comment (spolti) [inline kieserver/image.yaml:206]: maybe improve this to topic_name=topic_name.
see my comments on jboss-container-images/jboss-kie-modules#446

**Review by spolti — CHANGES_REQUESTED** [review]

File: kieserver/image.yaml (hunk 4 above)
Comment (desmax74) [reply kieserver/image.yaml:206]: Done

File: kieserver/image.yaml [hunk 5]
```diff
@@ -199,7 +199,31 @@ envs:
...
+    example: "message=topic_name"
+    description: "Contains the mapping message/signal=topicName for every topic that needs to be mapped globally, this env will be used by system prop org.kie.server.jbpm-kafka.ext.topics."
```
Comment (spolti) [inline kieserver/image.yaml:208]: mention here how user can provide more than one topic.

**Review by spolti — CHANGES_REQUESTED** [review]

File: kieserver/image.yaml (hunk 5 above)
Comment (desmax74) [reply kieserver/image.yaml:208]: ok

**Review by spolti — APPROVED** [review]

**Comment by spolti** [comment]
Hold until https://github.com/kiegroup/kie-cloud-operator/pull/614 and https://github.com/jboss-container-images/jboss-kie-modules/pull/446 get merged first.

File: kieserver/image.yaml [hunk 6]
```diff
@@ -199,7 +199,31 @@ envs:
     description: "Disable Maven pull dependencies for immutable KIE Server configurations for S2I and pre built kjars. Useful for pre-compiled kjar."
   - name: "KIE_SERVER_DISABLE_KC_VERIFICATION"
     example: "true"
-    description: "Disable Maven KIE Jar verification. It is recommended to test the kjar manually before disabling this verification."
+    description: "Disable Maven KIE Jar verification. It is recommended to test the kjar manually before disabling this verification."  
+  - name: "KIE_SERVER_KAFKA_EXT_ENABLED"
+    example: "true"
+    description: "Setting with true enable the others KIE_KAFKA_EXT environments used by Kafka, this env will be used by system prop org.kie.kafka.server.ext.disable with the opposite value"
```
Comment (spolti) [inline kieserver/image.yaml:205]:
```suggestion
    description: "Setting to true enables the other KIE Server Kafka environment variables used to configure kafka integration, this environment variable will be used to set the org.kie.kafka.server.ext.disable system property."
```

File: kieserver/image.yaml [hunk 7]
```diff
@@ -199,7 +199,31 @@ envs:
...
+  - name: "KIE_SERVER_KAFKA_EXT_TOPICS"
+    example: "message=topic_name"
+    description: "Contains the mapping message/signal=topicName for every topic that needs to be mapped globally, this env will be used by system prop org.kie.server.jbpm-kafka.ext.topics, multiple mappings must be provided separated by a comma"
```
Comment (spolti) [inline kieserver/image.yaml:208]:
```suggestion
    description: "Contains the mapping message/signal=topicName for every topic that needs to be mapped globally, this environment variable will be used to set the org.kie.server.jbpm-kafka.ext.topics system property, to provide more than one mapping use comma separated values."
```

File: kieserver/image.yaml [hunk 8]
```diff
@@ -199,7 +199,31 @@ envs:
...
+  - name: "KIE_SERVER_KAFKA_EXT_BOOTSTRAP_SERVERS"
+    example: "localhost:9092"
+    description: "A comma separated list of host/port pairs to use for establishing the initial connection to the Kafka cluster, this env will be used by system prop org.kie.server.jbpm-kafka.ext.bootstrap.servers"
```
Comment (spolti) [inline kieserver/image.yaml:211]:
```suggestion
    description: "A comma separated list of host/port pairs to use for establishing the initial connection to the Kafka cluster, this environment variable will be used to set the org.kie.server.jbpm-kafka.ext.bootstrap.servers system property."
```

File: kieserver/image.yaml [hunk 9]
```diff
@@ -199,7 +199,31 @@ envs:
...
+  - name: "KIE_SERVER_KAFKA_EXT_CLIENT_ID"
+    example: "applicationName"
+    description: "An id string to pass to the server when making requests. Not passed by default (the purpose of this property is to provide a logical application name for logging), this env will be used by system prop org.kie.server.jbpm-kafka.ext.client.id"
```
Comment (spolti) [inline kieserver/image.yaml:214]:
```suggestion
    description: "This configuration allows users to set an ID to provide a logical application name for logging purposes, not set by default.This environment variable will be used to set the org.kie.server.jbpm-kafka.ext.client.id system property."
```

File: kieserver/image.yaml [hunk 10]
```diff
@@ -199,7 +199,31 @@ envs:
...
+  - name: "KIE_SERVER_KAFKA_EXT_AUTOCREATE_TOPICS"
+    example: "true"
```
Comment (spolti) [inline kieserver/image.yaml:216]:
```suggestion
    example: "false"
```

File: kieserver/image.yaml [hunk 11]
```diff
@@ -199,7 +199,31 @@ envs:
...
+    description: "Allow automatic topic creation.Allow automatic topic creation. Enabled by default, this env will be used by system prop org.kie.server.jbpm-kafka.ext.allow.auto.create.topics"
```
Comment (spolti) [inline kieserver/image.yaml:217]:
```suggestion
    description: "Allow automatic topic creation. Enabled by default, this environment variable will be used to set the org.kie.server.jbpm-kafka.ext.allow.auto.create.topics system property."
```

File: kieserver/image.yaml [hunk 12]
```diff
@@ -199,7 +199,31 @@ envs:
...
+  - name: "KIE_SERVER_KAFKA_EXT_GROUP_ID"
+    example: "jbpm-consumer"
+    description: "A unique string that identifies the group this consumer belongs to. Default to 'jbpm-consumer' org.kie.server.jbpm-kafka.ext.group.id org.kie.server.jbpm-kafka.ext.group.id"
```
Comment (spolti) [inline kieserver/image.yaml:220]:
```suggestion
    description: "A unique string that identifies the group this consumer belongs to. Default to 'jbpm-consumer'. this environment variable will be used to set the org.kie.server.jbpm-kafka.ext.group.id system property."
```

File: kieserver/image.yaml [hunk 13]
```diff
@@ -199,7 +199,31 @@ envs:
...
+  - name: "KIE_SERVER_KAFKA_EXT_ACKS"
+    example: "1"
```
Comment (spolti) [inline kieserver/image.yaml:222]:
```suggestion
    example: "2"
```

File: kieserver/image.yaml [hunk 14]
```diff
@@ -199,7 +199,31 @@ envs:
...
+    description: "The number of acknowledgments the producer requires the leader to have received before considering a request complete. 1 by default, which means the leader will write the record to its local log but will respond without awaiting full acknowledgement from all followers, this env will be used by system prop org.kie.server.jbpm-kafka.ext.acks"
+  - name: "KIE_SERVER_KAFKA_EXT_MAX_BLOCK_MS"
+    example: "2000"
```
Comment (spolti) [inline kieserver/image.yaml:225]:
```suggestion
    example: "6000"
```

File: kieserver/image.yaml [hunk 15]
```diff
@@ -199,7 +199,31 @@ envs:
...
+  - name: "KIE_SERVER_KAFKA_EXT_AUTOCREATE_TOPICS"
+    example: "true"
+    description: "Allow automatic topic creation.Allow automatic topic creation. Enabled by default, this env will be used by system prop org.kie.server.jbpm-kafka.ext.allow.auto.create.topics"
+  - name: "KIE_SERVER_KAFKA_EXT_GROUP_ID"
+    example: "jbpm-consumer"
+    description: "A unique string that identifies the group this consumer belongs to. Default to 'jbpm-consumer' org.kie.server.jbpm-kafka.ext.group.id org.kie.server.jbpm-kafka.ext.group.id"
+  - name: "KIE_SERVER_KAFKA_EXT_ACKS"
+    example: "1"
+    description: "The number of acknowledgments the producer requires the leader to have received before considering a request complete. 1 by default, which means the leader will write the record to its local log but will respond without awaiting full acknowledgement from all followers, this env will be used by system prop org.kie.server.jbpm-kafka.ext.acks"
```
Comment (spolti) [inline kieserver/image.yaml:223]:
```suggestion
    description: "The number of acknowledgments that the producer requires the leader to have received before considering a request to be complete. The default value is 1, which means the leader will write the record to its local log but will respond without awaiting full acknowledgment from all followers, this environment variable will be used to set the org.kie.server.jbpm-kafka.ext.acks system property."
```

File: kieserver/image.yaml [hunk 16]
```diff
@@ -199,7 +199,31 @@ envs:
...
+  - name: "KIE_SERVER_KAFKA_EXT_MAX_BLOCK_MS"
+    example: "2000"
+    description: "Number of milliseconds that indicates how long publish method will block. Default 2000 (2 seconds), this env will be used by system prop org.kie.server.jbpm-kafka.ext.max.block.ms"
```
Comment (spolti) [inline kieserver/image.yaml:226]:
```suggestion
    description: "Value in milliseconds that indicates how long the 'publish' method will block the operation. Default 2000 milliseconds (2 seconds), this environment variable will be used to set the org.kie.server.jbpm-kafka.ext.max.block.ms system property."
```

**Review by spolti — APPROVED** [review]

**Comment by spolti** [comment]
Hold until kiegroup/kie-cloud-operator#614 and jboss-container-images/jboss-kie-modules#446 get merged first.

**Review by Kaitou786 — APPROVED** [review]

**Comment by spolti** [comment]
@desmax74 can you please rebase?

**Comment by spolti** [comment]
before, remember to fetch the changes I made on this PR.

File: kieserver/image.yaml [hunk 17]
```diff
@@ -206,6 +206,30 @@ envs:
   - name: "KIE_SERVER_JBPM_CLUSTER_TRANSPORT_LOCK_TIMEOUT"
     example: "70000"
     description: "Jbpm Cache transport lock timeout, value in milliseconds. The default value is 60000 ms."
+  - name: "KIE_SERVER_KAFKA_EXT_ENABLED"
+    example: "true"
+    description: "Setting to true enables the other KIE Server Kafka environment variables used to configure kafka integration, this environment variable will be used to set the org.kie.kafka.server.ext.disable system property."
+  - name: "KIE_SERVER_KAFKA_EXT_TOPICS"
+    example: "message=topic_name"
```
Comment (spolti) [inline kieserver/image.yaml:213]: message=topic_name
```suggestion
    example: "message=topic_name, message2=topic2"
```

File: kieserver/image.yaml [hunk 18]
```diff
@@ -206,6 +206,30 @@ envs:
...
+    description: "Contains the mapping message/signal=topicName for every topic that needs to be mapped globally, this env will be used by system prop org.kie.server.jbpm-kafka.ext.topics, multiple mappings must be provided separated by a comma"
```
Comment (spolti) [inline kieserver/image.yaml:214]:
```suggestion
    description: "Contains the mapping message/signal=topicName for every topic that needs to be mapped globally, this environment variable will be used to set the org.kie.server.jbpm-kafka.ext.topics system property, to provide more than one mapping use comma separated values."
```

**Review by spolti — APPROVED** [review]

**Pull request merged**